bootable-jar/WFLY-14411_Bootable_Jar_Upgrade.adoc

```diff
@@ -0,0 +1,246 @@
+
+* mailto:{jkasik@redhat.com}[Jan Kasik]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [X] QE
+
+=== Affected Projects or Components
+
+* Bootable JAR Maven plugin
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [ ] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+* The upgrade mechanism is done from the Bootable JAR Maven plugin.
+
+* Components that we want to upgrade are referenced with Maven Coordinates 
+(GroupId, ArtifactId, Classifier, Version and Type) from the Maven dependencies section of the Maven project.
+That is the nominal Maven way of expressing dependencies.
+
+* We are not automatically using all artifacts present in the project dependencies section for the upgrade.  
+A user needs to list the GroupId/ArtifactId/[Classifier] of the artifacts he wants to see used during the upgrade.
+
+* The `wildfly-jar-maven-plugin` configuration is evolved with an `<overridden-server-artifacts>` set that contains the artifacts that 
+are in use for the upgrade. These artifacts are upgraded globally. Any feature-pack that references it will be upgraded.
+
+* The `wildfly-jar-maven-plugin` `<FeaturePack>` configuration element is evolved with an `<overridden-artifacts>` set that contains the artifacts (JBoss Modules module 
+artifacts only) that are in use for the upgrade. These artifacts are upgraded only in the feature-pack that define them. For example we could have FP1 and FP2 that depend on
+the same GA but with different versions. In such a case we want to control which actual artifact is impacted by the upgrade. 
+
+* The overridden artifacts inside  the `<FeaturePack>` configuration element can define a version. This allows to have 
+multiple artifacts with the same GA to be upgraded to different versions according to the feature-pack that contain them.
+
+* JBoss Modules jar artifacts and Galleon feature-packs zip artifacts are treated the same way and can be present in the `overriden-server-artifacts` set.
+
+=== Hard Requirements
+
+* This RFE doesn't introduce backward incompatibility.
+
+* This RFE is made possible starting with WildFly 23. Older WildFly release can't be used.
+
+* Usage of Galleon feature-pack location (FPL) to reference the Galleon channel of a producer registered inside a universe) is still possible.
+
+* This mechanism strongly depends on Maven dependency resolution. 
+Any artifact to upgrade must be present in accessible Maven repositories (remote or local cache).
+
+* An artifact present in the  `overridden-server-artifacts` or `overridden-artifacts` set must be present in the dependencies section. 
```

Comment:
I believe this change of element name is a breaking change vs the plugin version just released with WF 23. Can this rename be avoided? Or at least the old name works but is undocumented?

Reply:
This was a typo in the spec, the overridden-server-artifacts is the correct name used with Maven plugin 4.0.0.Final/WF23.